remediation/fence-agents-remediation/fence-agents-remediation.md

```diff
@@ -0,0 +1,29 @@
+layout: default
+title: Fence Agents Remediation
+nav_order: 3
+has_children: true
+parent: Remediation
+---
+
+## Fence Agents Remediation
+
+Generally Available
+{: .label .label-green }
+
+[Fence Agents Remediation](https://github.com/medik8s/fence-agents-remediation#readme) (FAR) is a Kubernetes operator that uses external tools to *fence* unhealthy nodes. The tools are a set of well-known [upstream fence agents (FA)](https://github.com/ClusterLabs/fence-agents), where each one of them can be used for different environments, to fence a node using a reboot action. Moreover, FAR can be used by the [Node HealthCheck Operator](https://github.com/medik8s/node-healthcheck-operator#readme) (or any other operator that supports the external remediation API) to remediate an unhealthy node.
+
+FAR is not only fencing a node when it becomes unhealthy, it also *remediates* the node from being unhealthy to healty again. It adds a taint to evict stateless pods, fence the node with a fence agent and fully remediate it with resource deletion to remove any remaining workloads (mostly statefull workloads) after reboot.
+
+FAR is recommended when a node becomes unhealthy, and we want to completely isolate the node from a cluster, since we can’t “trust” the unhealthy node, to prevent it from accessing the shared resources like [RWO volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes). The operator is available in the Kubernetes community, [OperatorHub.io](https://operatorhub.io/operator/fence-agents-remediation), and can be installed manually as mentioned in [FAR documentation](https://github.com/medik8s/fence-agents-remediation#installation)
+
+As opposed to other remediation systems, FAR requires credentials/access to a management interface (e.g., Intelligent Platform Management Interface ([IPMI](https://en.wikipedia.org/wiki/Intelligent_Platform_Management_Interface)), etc.) or an API to reboot a node, but it has many advantages.
+
+### Advantages
+
+* Robustness - FAR can remediate an unhealthy node using a management interface (e.g., IPMI) while still keeping control plane connectivity.
+* Speed - FAR is fast as it adds the FAR NoExecute taint, and deletes workloads to accelerate workload rescheduling, in addition to rebooting the node using a fence agent.
+* Diversity - FAR includes several FAs from a whole known set of upstream fencing agents for bare metal servers, virtual machines, cloud platforms, etc.
+* Adjustability - FAR allows to set up different parameters to access the management interface that remediates the node.
+* Compatibility with NHC - FAR works with NHC to provide high availability for nodes in an automatic manner. NHC detects the node’s health and triggers FAR to remediate the node until it becomes healthy again.
+* Standalone - FAR can be used as a standalone solution to remediate nodes with an administrator's help. Moreover, FAR is not limited to NHC and could work with another detection mechanism.
```

Comment:
What does "standalone" mean?
You can create the remediation CRs yourself, or use other detection mechanism, for all our remediators.
SNR can even remediate without any interaction 🤷🏼‍♂️

Reply:
> You can create the remediation CRs yourself, or use other detection mechanism, for all our remediators.

Correct, I was thinking of expressing it so users won't think that FAR, similar to SNR and MDR, is limited to be used with NHC. I thought it would help to increase engagement in case users want to pre-automate the remediation manually or with another detecting mechanism to remediate a node.

It might better to use 'Semi-automation'.

>Semi-automation - FAR, and all the remediation operators from Medik8s, can semi-automate the node remediation once the node has been detected/selcted to be remediated. The selection can be with NHC, or another detection mechanism, or even an administrator's help.

Reply:
I have decided to rephrase it again in a new paragraph prior to the advantages section